ui/src/components/identities/IdentityForm.tsx

```diff
@@ -82,13 +114,20 @@ export function IdentityForm({
     return aborter;
   }, [fetchNetworks]);
 
-  if (hasAsyncTaskFailed(supportedNetworks)) {
+  if (
+    hasAsyncTaskFailed(supportedNetworks) ||
+    (isAsyncTaskDataAvailable(supportedNetworks) && !supportedNetworks.data.length)
+  ) {
+    const errorMessage = hasAsyncTaskFailed(supportedNetworks)
+      ? supportedNetworks.error.message
+      : "";
```

Comment:
An error msg like `""` is illegal. We must model this better.

Reply:
I'd go this way:

```typescript
switch (supportedNetworks.status) {
  case "pending":
  case "loading": {
    return (
      <Card className="centered">
        <LoadingResult />
      </Card>
    );
  }
  case "failed": {
    return (
      <Card className="centered">
        <ErrorResult
          error={[
            "An error occurred while downloading the supported networks from the API:",
            supportedNetworks.error.message,
          ].join("\n")}
        />
      </Card>
    );
  }
  case "reloading":
  case "successful": {
    if (supportedNetworks.data.length === 0) {
      return (
        <Card className="centered">
          <ErrorResult error="No supported networks found" />
        </Card>
      );
    }
   ....
  }
```

Reply:
In reality, checking `supportedNetworks.data.length` is not even a job of the view, but of the parser.

Since we're already doing that check in the type `[Network, ...Network[]]` and therefore in the parser, this check is not necessary here. Whenever we get a 200 and the list is empty we will get a zod error in the view.

Please, keep the switch/case statement for clarity.